srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditModelTerm.java

```diff
@@ -0,0 +1,696 @@
+            throws InvalidIdentifierException;
+    }
+    
+    private class ConstantTypeState implements TypeState
+    {
+        @Override
+        public TermType getAssociatedType()
+        {
+            return TermType.CONSTANT;
+        }
+        
+        @Override
+        public ModelTermSummary makeSummary(final ModelInspectionService modelService)
+        {
+            return new ModelTermSummary("Constant", "", fCoefficient);
+        }
+        
+        @Override
+        public BaseValidator getValidator(final ModelInspectionService modelService)
+        {
+            return new ConstantTermValidator();
+        }
+
+        @Override
+        public ConstantTerm buildTerm(final ModelInspectionService modelService)
+        {
+            return new ConstantTerm(fCoefficient);
+        }
+    }
+    
+    private class VariableTypeState implements TypeState
+    {
+        private class TermBuilder extends ExceptionlessVariableVisitor
+        {
+            private SingleVariableTerm fBuiltTerm;
+            
+            public SingleVariableTerm getBuiltTerm()
+            {
+                return fBuiltTerm;
+            }
+
+            @Override
+            public void visitProcedure(final ProcedureVariable variable)
+            {
+                fBuiltTerm = new ProcedureTerm(variable, fCoefficient);
+            }
+            
+            @Override
+            public void visitNumerical(final NumericalVariable variable)
+            {
+                fBuiltTerm = new NumericalTerm(variable, fCoefficient);
+            }
+            
+            private DiscreteTerm buildDiscreteTerm(final DiscreteVariable variable)
+            {
+                final int index = variable.getOptions().indexOf(
+                        // Just construct an equivalent option to find the index.
+                        new MultiSelectOption(fOptionValue));
+
+                if (index != -1)
+                {
+                    return new DiscreteTerm(variable, index, fCoefficient);
+                }
+                else
+                {
+                    throw new IllegalStateException(
+                            "no such option value in variable " + variable);
+                }
+            }
+            
+            @Override
+            public void visitMultiSelect(final MultiSelectVariable variable)
+            {
+                fBuiltTerm = buildDiscreteTerm(variable);
+            }
+            
+            @Override
+            public void visitDiscreteNumerical(final DiscreteNumericalVariable variable)
+            {
+                fBuiltTerm = buildDiscreteTerm(variable);
+            }
+            
+            @Override
+            public void visitBoolean(final BooleanVariable variable)
+            {
+                fBuiltTerm = new BooleanTerm(variable, fCoefficient);
+            }
+        };
+        
+        private class TermTypeTranslator extends ExceptionlessVariableVisitor
```

Comment:
I think this class could be more accurately named VariableTermTypeTranslator or something similar since it doesn't cover all of the term types, on the ones related to variables.